lib/view_model/dashboard/home_settings_view_model.dart

```diff
@@ -113,6 +116,13 @@ abstract class HomeSettingsViewModelBase with Store {
     } finally {
       isAddingToken = false;
     }
+
+    if (_balanceViewModel.wallet.type == WalletType.zano) {
+      await zano!.addZanoAssetById(_balanceViewModel.wallet, contractAddress);
+    }
+
+    _updateTokensList();
+    _updateFiatPrices(token);
```

Comment:
still duplicate, just move the if condition below the other if conditions and remove the duplicate update functions